src/praisonai-agents/tests/unit/test_gateway_emergency_stop.py

```
@@ -0,0 +1,127 @@
+    EmergencyStopConfig,
+    EmergencyStopProtocol,
+    EmergencyStopState,
+    FileEmergencyStop,
+    GatewayConfig,
+    NullEmergencyStop,
+)
+
+
+def test_null_and_file_conform_to_protocol():
+    assert isinstance(NullEmergencyStop(), EmergencyStopProtocol)
+    assert isinstance(FileEmergencyStop("/tmp/x.pause"), EmergencyStopProtocol)
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/x.pause"

(S108)